app/src/cli/package/mod.rs

```diff
@@ -0,0 +1,174 @@
+use std::{
+    net::IpAddr,
+    path::{Path, PathBuf},
+};
+
+use ambient_native_std::asset_cache::AssetCache;
+use ambient_package::PackageId;
+use anyhow::Context;
+use clap::{Args, Subcommand};
+
+use super::PackagePath;
+
+pub mod build;
+pub mod deploy;
+pub mod new;
+pub mod run;
+pub mod serve;
+
+#[derive(Subcommand, Clone, Debug)]
+/// Package-related commands.
+pub enum Package {
+    /// Regenerate the ID for a given package to make it compliant with the ID scheme.
+    RegenerateId {
+        #[command(flatten)]
+        package: PackageArgs,
+    },
+}
+impl Package {
+    pub fn args(&self) -> &PackageArgs {
+        match self {
+            Package::RegenerateId { package } => package,
+        }
+    }
+}
+
+#[derive(Args, Clone, Debug)]
+pub struct PackageArgs {
+    /// Dummy flag to catch Rust users using muscle memory and warn them
+    #[arg(long, short, hide = true)]
+    pub project: bool,
+
+    /// The path or URL of the package to run; if not specified, this will default to the current directory
+    pub path: Option<String>,
+
+    /// Build all the assets with debug information; this will make them less performant and larger but easier to debug (default for all commands apart from `deploy` and `serve`)
+    #[arg(long, conflicts_with = "release")]
+    debug: bool,
+
+    /// Build all the assets with full optimization; this will make them faster and smaller but more difficult to debug (default for `deploy` and `serve`)
+    #[arg(short, long)]
+    release: bool,
+
+    /// Avoid building the package
+    #[arg(long)]
+    pub no_build: bool,
+
+    #[arg(long)]
+    /// Perform a clean build
+    pub clean_build: bool,
+
+    #[arg(long)]
+    /// Only build the WASM modules
+    pub build_wasm_only: bool,
+}
+impl PackageArgs {
+    pub fn is_release(&self) -> Option<bool> {
+        match (self.debug, self.release) {
+            (true, false) => Some(false),
+            (false, true) => Some(true),
+            (false, false) => None,
+            (true, true) => {
+                // clap's conflict_with should prevent this from happening
+                panic!("debug and release are mutually exclusive")
+            }
+        }
+    }
+
+    pub fn package_path(&self) -> anyhow::Result<PackagePath> {
+        self.path.clone().try_into()
+    }
+}
+
+#[derive(Args, Clone, Debug)]
+pub struct HostCli {
```

Comment:
I'm assuming this one is left here because of the hurdles.

Reply:
Yeah, that's just the common args for the commands that host (run, serve). Can't do much about that, I think.